spdx/parsers/jsonyamlxml.py

```diff
@@ -658,7 +658,7 @@ def parse_snippet_license_info_from_snippet(self, license_info_from_snippet):
                         self.value_error("SNIPPET_LIC_INFO", lic_in_snippet)
                 else:
                     self.value_error("SNIPPET_LIC_INFO", lic_in_snippet)
-        else:
+        elif license_info_from_snippet is not None:
```

Comment:
If `license_info_from_snippet` is `None` this method is called up unnecessarily. I would suggest to first check in `parse_snippet` if there is a value and if not don't call the method. This applies for all the fields that are mandatory now. I know that this hasn't been done before, so for e.g. packages this would apply as well. I'm not sure if we should change everything in this PR but mybe we could at least start with the parts that are touched here anyway.

Reply:
I agree that the `None` check is quite hidden there but as I only copied this behaviour from already established cases I propose to leave this refactoring to the big restructure that is coming to the parsers.

Reply:
I tend to agree to leave larger refactorings (that should be applied to most of the parser in this case) for later